config_parser.cc

```diff
@@ -149,6 +152,8 @@ NginxConfigParser::TokenType NginxConfigParser::ParseToken(std::istream* input,
 
 bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
   std::stack<NginxConfig*> config_stack;
+  std::stack<TokenType> block_balancer;  //matches block start tokens with block end tokens 
```

Comment:
I believe this can be done with just an int counter since all you're doing is adding and subtracting elements to the stack to keep track of the count (stack accomplishes the same thing, it just adds a bit more overhead!)